mm2src/mm2_main/tests/docker_tests/eth_docker_tests.rs

```diff
@@ -842,21 +1048,51 @@ fn send_and_spend_erc1155_maker_payment() {
         maker_pub: &maker_global_nft.parse_pubkey(&maker_pubkey).unwrap(),
         swap_unique_data: &[],
         contract_type: &ContractType::Erc1155,
-        swap_contract_address: &nft_swap_contract(),
+        swap_contract_address: &sepolia_etomic_maker_nft(),
     };
     let spend_tx = block_on(taker_global_nft.spend_nft_maker_payment_v2(spend_payment_args)).unwrap();
+    log!(
+        "Taker spent ERC1155 NFT Maker payment, tx hash: {:02x}",
+        spend_tx.tx_hash()
+    );
 
     let confirm_input = ConfirmPaymentInput {
         payment_tx: spend_tx.tx_hex(),
         confirmations: 1,
         requires_nota: false,
-        wait_until: now_sec() + 60,
+        wait_until: now_sec() + 80,
+        check_every: 1,
+    };
+    taker_global_nft.wait_for_confirmations(confirm_input).wait().unwrap();
+
+    let taker_address = block_on(taker_global_nft.my_addr());
+    let balance = erc1155_balance(taker_address, U256::from(1));
+    assert_eq!(balance, U256::from(3));
```

Comment:
> can we also test swapping only a portion of the coins/nfts if that's supported in 1155?

could you clarify please do you mean 0.5 or transfer 2 of 3 owned nfts? While erc155 standard allows holding multiple tokens (copies of the same item), each token itself is indivisible.

some inside from my nft travel, I started to use `amount: BigDecimal`, as I thought that one day nft feature reaches swaps milestone, so I will need to integrate nft swaps into legacy swap protocol (we didnt have swap v2 in plans).
However, as we have right now swap protocol upgrade v2 in wip and have the idea to implement state machine for nft specific states we could start to use `BigUint` instead of `BigDecimal` for nft amount. will see during nft swap implementation.

Reply:
Yup, meant to first send (for e.g.) `2` out of `3` to the taker, make sure the taker has `2` and the maker has `1`.

Reply:
> Yup, meant to first send (for e.g.) `2` out of `3` to the taker, make sure the taker has `2` and the maker has `1`.

got you, updated tests in this https://github.com/KomodoPlatform/komodo-defi-framework/pull/2129/commits/d9980622ac00f812bd23ef60b120234de0ab765c commit
it is in #2129 PR